plugins/woocommerce-blocks/assets/js/blocks/product-filter/inner-blocks/price-filter/frontend.ts

```diff
@@ -61,60 +102,19 @@ store< PriceFilterStore >( 'woocommerce/product-filter-price', {
 	},
 	actions: {
 		updateProducts: ( event: HTMLElementEvent< HTMLInputElement > ) => {
-			const { decimalSeparator } = getCurrency();
 			const context = getContext< PriceFilterContext >();
-			const { minRange, minPrice, maxPrice, maxRange } = context;
-			const type = event.target.name;
-			const value = parseInt(
-				event.target.value
-					.replace(
-						new RegExp( `[^0-9\\${ decimalSeparator }]+`, 'g' ),
-						''
-					)
-					.replace(
-						new RegExp( `\\${ decimalSeparator }`, 'g' ),
-						'.'
-					),
-				10
-			);
-
-			const currentMinPrice =
-				type === 'min'
-					? Math.min(
-							Number.isNaN( value ) ? minRange : value,
-							maxPrice
-					  )
-					: minPrice;
-			const currentMaxPrice =
-				type === 'max'
-					? Math.max(
-							Number.isNaN( value ) ? maxRange : value,
-							minPrice
-					  )
-					: maxPrice;
 
 			// In some occasions the input element is updated with the incorrect value.
 			// By using the element that triggered the event, we can ensure the correct value is used for the input.
 			const element = getElement();
-			if ( type === 'min' ) {
-				element.ref.value = currentMinPrice;
-			}
 
-			if ( type === 'max' ) {
-				element.ref.value = currentMaxPrice;
+			debounceUpdate( context, element, event );
```

Comment:
I used debounce inside the function rather than wrapping the whole function to ensure that the context (`getContext()`) is captured immediately when `updateProducts` is triggered. If I debounce the entire function, `context` might be `undefined` when the debounced function runs due to the delay. This approach ensures that the valid context is passed into the debounced logic.